### PR TITLE
chore: optimize test pipeline

### PR DESCRIPTION
**Pull request by cofin**
[//]: # "By submitting this pull request, you agree to:"
[//]: # "- follow [Jolt's Code of Conduct](https://github.com/jolt-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow [PSFs's Code of Conduct](https://www.python.org/psf/conduct/)"
[//]: # "- follow [Jolt's contribution guidelines](https://github.com/jolt-org/.github/blob/main/CONTRIBUTING.md)"

### Pull Request Checklist

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [ ] Pre-Commit Checks were ran and passed
- [ ] Tests were ran and passed

### Description
[//]: # "Please describe your pull request for new release changelog purposes"

This PR allows the database infrastructure to be started outside of the test pipeline.

It also updates the CI actions to start the containers before executing tests.

### Close Issue(s)
[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

-

### CODE REVIEW

**Review by JacobCoffee — APPROVED** [review]

**Comment by cofin** [comment]
Doesn't really feel optimized :)

**Comment by cofin** [comment]
I think there's been some sort of regression with `asyncmy` causing quite a bit of slowdown

File: tests/docker_service_fixtures.py [hunk 1]
```diff
@@ -110,7 +110,7 @@ def stop(self, name: str) -> None:
 
     def down(self) -> None:
         if not SKIP_DOCKER_COMPOSE:
-            self.run_command("down", "--remove-orphans", "--volumes", "-t", "10")
+            self.run_command("down", "-t", "10")
```
Comment (provinzkraut) [inline tests/docker_service_fixtures.py:113]: Won't this potentially cause problems on subsequent runs locally when `--remove-orphans` isn't used?

Comment (cofin) [reply tests/docker_service_fixtures.py:113]: I didn't want to remove that, but I did as a test to see if it fixes the issues I'm seeing. 

The tests just appear to hang around 88% in CI and around 92% when I run it locally. 

I'll add it back as I don't think it's causing this issue.

**Pull request closed without merging**